File: Leetcode/Python/0009_Palindrome_Number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        s = str(x)
        l = len(s)
        if l <= 1:
            return True
        x = 0
        y = l-1
        while x < y:
            if s[x] != s[y]:
                return False
            else:
                x += 1
                y -= 1
        return True

File: Leetcode/Java/0009_Palindrome_Number.java
class Solution {
    public boolean isPalindrome(int x) {
        int origin = x;
        if (x < 0) {
            return false;
        }
        int reversed = 0;
        while (x != 0) {
            int t = x % 10;
            reversed = reversed * 10 + t;
            x = x / 10;
        }
        if (origin == reversed) {
            return true;
        } else {
            return false;
        }
    }
}